docs/resources/Guild.md

```diff
@@ -280,18 +280,19 @@ A partial [guild](#DOCS_RESOURCES_GUILD/guild-object) object. Represents an Offl
+| avatar?                       | ?string                                         | the member's [guild avatar hash](#DOCS_REFERENCE/image-formatting)                                                                                                                                                                   |
+| roles                         | array of snowflakes                             | array of [role](#DOCS_TOPICS_PERMISSIONS/role-object) object ids                                                                                                                                                                     |
+| joined_at                     | ISO8601 timestamp                               | when the user joined the guild                                                                                                                                                                                                       |
+| premium_since?                | ?ISO8601 timestamp                              | when the user started [boosting](https://support.discord.com/hc/en-us/articles/360028038352-Server-Boosting-) the guild                                                                                                              |
+| deaf                          | boolean                                         | whether the user is deafened in voice channels                                                                                                                                                                                       |
+| mute                          | boolean                                         | whether the user is muted in voice channels                                                                                                                                                                                          |
+| pending?                      | boolean                                         | whether the user has not yet passed the guild's [Membership Screening](#DOCS_RESOURCES_GUILD/membership-screening-object) requirements                                                                                               |
+| permissions?                  | string                                          | total permissions of the member in the channel, including overwrites, returned when in the interaction object                                                                                                                        |
+| communication_disabled_until? | ?ISO8601 timestamp                              | when the user's [timeout](https://support.discord.com/hc/en-us/articles/4413305239191-Time-Out-FAQ) will expire and the user will be able to communicate in the guild again, null or a time in the past if the user is not timed out |
```

Comment:
When can this property be missing?

Reply:
It seems to be on every guild member object.
Had no case yet, where it was missing.

Reply:
oh this is a good point - @hemu is it optional?

Reply:
ah to be honest the optionality of fields from the perspective of the public api is always a tricky question to answer definitively. b/c the objects we define in the docs are defined more for convenience, so they can be referred to from other places in the docs rather than something that always maps 1:1 to how we use models internally. I think the best answer I can give is when we serialize the guild member model, we currently always seem to include the `communication_disabled_until` field. This doesn't 100% guarantee it is always included unless we thoroughly check every single endpoint where this model is included in some way. But I *think* that means we can remove the optional?

Reply:
since from a pure data model perspective `communication_disabled_until` isn't a core field for the member model like `joined_at` is, how about we leave it as optional just in case, since I'd imagine the opposite -- assuming it is required now and at some point in the future an endpoint whose document references this Guild Member Object in docs fails to include this field -- is a worse developer experience. What do you think?

Reply:
We should leave it as optional :)